### PR TITLE
checkIfItemsQuantityIsEqualOrHigherThanItemTotalQuantityDiscount

### PR DESCRIPTION
**Pull request by p-zielinski**

### CODE REVIEW

File: src/api-extension/utils/checkIfItemsQuantityIsEqualOrHigherThanItemTotalQuantityDiscount.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { LineItem } from '@commercetools/platform-sdk';
+
+export default function checkIfItemsQuantityIsEqualOrHigherThanItemTotalQuantityDiscount(
+  lineItems: LineItem[],
+): boolean {
+  return !!lineItems?.find((lineItem) => {
+    if (lineItem.custom?.fields?.applied_codes) {
+      const { quantity: itemQuantity } = lineItem;
+      const appliedCodes = (lineItem.custom?.fields?.applied_codes ?? [])
```
Comment (tomek-lochmanczyk-rspective) [inline src/api-extension/utils/checkIfItemsQuantityIsEqualOrHigherThanItemTotalQuantityDiscount.ts:9]: you can probably shorten it to:
`const totalQuantityDiscount = (lineItem.custom?.fields?.applied_codes ?? [])
        .map((code) => JSON.parse(code))
        .filter((code) => code.type === 'UNIT')
        ,filter((code) => code.quantity)
        .reduce((sum, codeObject) => sum += codeObject.quantity, 0);`

File: src/api-extension/cart.service.ts [hunk 2]
```diff
@@ -452,13 +430,16 @@ export class CartService {
...
       return this.setCustomTypeForInitializedCart();
     }
+
     if (
-      this.checkIfQuantityIsEqualOrHigherThanTotalQuantityDiscount(
+      checkIfItemsQuantityIsEqualOrHigherThanItemTotalQuantityDiscount(
         cart.lineItems,
       )
     ) {
+      console.log('a');
```
Comment (tomek-lochmanczyk-rspective) [inline src/api-extension/cart.service.ts:439]: remove console log

**Review by tomek-lochmanczyk-rspective — APPROVED** [review]

File: src/api-extension/utils/checkIfItemsQuantityIsEqualOrHigherThanItemTotalQuantityDiscount.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { LineItem } from '@commercetools/platform-sdk';
+
+export default function checkIfItemsQuantityIsEqualOrHigherThanItemTotalQuantityDiscount(
+  lineItems: LineItem[],
+): boolean {
+  return !!lineItems?.find((lineItem) => {
+    if (lineItem.custom?.fields?.applied_codes) {
```
Comment (jkaliszuk) [inline src/api-extension/utils/checkIfItemsQuantityIsEqualOrHigherThanItemTotalQuantityDiscount.ts:7]: Maybe us exit early pattern?
```
if(!lineItem.custom?.fields?.applied_codes.length){
  return true;
}
(...)
```

**Review by jkaliszuk — APPROVED** [review]
Consider my comment.

**Pull request merged**